pandas/core/indexing.py

```diff
@@ -2083,8 +2086,18 @@ def _setitem_with_indexer_missing(self, indexer, value):
                     # We get only here with loc, so can hard code
                     return self._setitem_with_indexer(new_indexer, value, "loc")
 
-            # this preserves dtype of the value
-            new_values = Series([value])._values
+            # this preserves dtype of the value and of the object
+            if isna(value == value):
```

Comment:
`value == value` to see if it's `pd.NA`? Not exactly clear why value is being compare to itself

Reply:
Yep, is there a better way to achieve this?

Reply:
I think `isna(value)` should just work right?

```
In [1]: pd.isna(pd.NA)
Out[1]: True

In [2]: pd.isna(np.nan)
Out[2]: True
```

Reply:
I don’t want to run in there when I get nan, because nan does not fit into int64 for example

Reply:
I see. Well If you only want to check for `pd.NA` I think `value is pd.NA` is clearer IMO

Reply:
We probably want to limit this to certain cases. For example, setting pd.NaT into a non-datetimelike series (eg float) results in object dtype when there is no enlargement, and I suppose we should strive to have the result consistent regardless of enlargement or not. 
The `is_valid_na_for_dtype` helper can maybe help with this.

Reply:
I like your idea of checking nans more broadly. So currently, if we are setting nan into Int64 it gets converted to pd.NA?

Reply:
Yes, for example:

```
In [7]: s = pd.Series([1, 2, 3], dtype="Int64")

In [8]: s[0] = np.nan

In [9]: s
Out[9]: 
0    <NA>
1       2
2       3
dtype: Int64
```

Reply:
Wasn’t aware of this. Will adjust accordingly. You are correct, this should be consistent

Reply:
Could you have another look @jorisvandenbossche? I tried to add relevant cases for enlargement and non-enlargement to ensure consistency. Let me know if there is something missing.

There is one open case:

```
ser = pd.Series([1, 2], dtype="Int64")
ser[1] = "a"
```
This raises while expansion casts to object
```
ser = pd.Series([1, 2], dtype="Int64")
ser[2] = "a"
```

This is true for ``rhs="a"`` and ``rhs=pd.NaT``. With non-ea dtypes we are casting to object. Do we want to be consistent here or is the difference intended?